File: ROUND_1/JAVA/Brute_force/Vault_security.java
package ROUND_1.JAVA.Brute_force;

import java.io.*;
import java.util.*;


public class Vault_security {
public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        if (n < 0) {
            System.out.println("NO");
            return;
        }

        String s = Integer.toString(n);

        int left = 0;
        int right = s.length() - 1;

        while (left < right) {

            if (s.charAt(left) != s.charAt(right)) {
                System.out.println("NO");
                return;
            }

            left++;
            right--;
        }

        System.out.println("YES");
    }

    
}
/*
Time Complexity
O(d) where d is the number of digits.
Space Complexity
O(d)
*/

File: ROUND_1/CPP/Brute_force/Vault_security.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    long long n;
    cin >> n;

    // Negative Vault IDs are never valid
    if (n < 0) {
        cout << "NO" << endl;
        return 0;
    }

    string s = to_string(n);

    int left = 0;
    int right = s.length() - 1;

    while (left < right) {

        if (s[left] != s[right]) {
            cout << "NO" << endl;
            return 0;
        }

        left++;
        right--;
    }

    cout << "YES" << endl;

    return 0;
}
/*
Time Complexity: O(d) where d is the number of digits.
Space Complexity: O(d)
*/
